### DISCUSSION TITLE
如何在前端程序调用 1Panel 的 API 接口

### DISCUSSION THREAD

**Discussion by ikxin** [1Panel-dev/1Panel#4120] [Q&A]
### 1Panel 版本

v1.10.0-lts

### 请描述您的需求或者改进建议

我想在其他的前端程序调用 1Panel 的接口，但是会遇到 CORS 跨域问题，请问可以在后端的接口加上 Access-Control-Allow-Origin: * 这个 header 吗，会不会产生安全问题？

如果会的话，那是否可以考虑开发一个功能，可以自定义配置白名单，允许指定的地址请求API接口。

另外，我看到现在的前后端分离用的是 cookie 来鉴权，后期是否有计划使用 jwt，这个好像用在前端更方便调用。

以上是我的一些想法和建议，希望开发者有时间可以看一下，愿 1Panel 越来越好，引领国内开源环境的发展！

**Comment by zhangyz189** [comment]
jwt目前是支持的
`
curl 'http://127.0.0.1:7001/api/v1/auth/login'  --data-binary '{"name":"admin_test","password":"admin_test1234","ignoreCaptcha":true,"captcha":"","captchaID":"nY8Cqeut3TjZMfJMAz0k","authMethod":"jwt","language":"zh"}' -H 'EntranceCode: emhhbmd5eg=='
`
`
curl 'http://127.0.0.1:7001/api/v1/dashboard/current/all/all' -H 'PanelAuthorization: eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJJRCI6MCwiTmFtZSI6ImFkbWluX3Rlc3QiLCJCdWZmZXJUaW1lIjozNjAwLCJpc3MiOiIxUGFuZWwiLCJleHAiOjE3MDkyODg4MDl9.pdknJdjLY4Fp8wCE9Gvaiic2rLoSdvUSJB9ossyya_I'
`

**Reply by AmberKod** [reply]
目前这种方式也不行了，会报错：1Panel Login Failed: Login API Error: ErrCaptchaCode   一定要走验证码了？

**Comment by ikxin** [comment]
> jwt目前是支持的 `curl 'http://127.0.0.1:7001/api/v1/auth/login' --data-binary '{"name":"admin_test","password":"admin_test1234","ignoreCaptcha":true,"captcha":"","captchaID":"nY8Cqeut3TjZMfJMAz0k","authMethod":"jwt","language":"zh"}' -H 'EntranceCode: emhhbmd5eg=='` `curl 'http://127.0.0.1:7001/api/v1/dashboard/current/all/all' -H 'PanelAuthorization: eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJJRCI6MCwiTmFtZSI6ImFkbWluX3Rlc3QiLCJCdWZmZXJUaW1lIjozNjAwLCJpc3MiOiIxUGFuZWwiLCJleHAiOjE3MDkyODg4MDl9.pdknJdjLY4Fp8wCE9Gvaiic2rLoSdvUSJB9ossyya_I'`

感谢回复，但 jwt 好像仍有 cors 跨域问题

**Reply by aaron931** [reply]
这个跨域问题，有办法解决么

**Reply by ikxin** [reply]
暂时还没有办法，等待官方后续跟进优化吧

**Reply by MillerFour** [reply]
请问下 EntranceCode  这个参数值如何获取？

**Reply by ikxin** [reply]
安全入口 base64 编码

**Comment by wanghe-fit2cloud** [comment]
建议使用密钥调用 API https://github.com/1Panel-dev/1Panel/discussions/10335#discussioncomment-15314747
